### PR TITLE
vdk-smarter: introduce simple open ai plugin

### PR DESCRIPTION
**Pull request by tozka**
Very simple plugin that reviews every query and scores it using OpenAI models. The purpose is to provide easy visibility into queries and their quality for the user. Though technically they could potentially use chatGPT themselves or static code analysis, that is less likely and and harder for them

A) because some queries are generated dynamically
B) Many queries could be part of a job and take (development) time to check. 
C) the prompt engineering is something that takes time as well and tuning as well. The prompts could be tuned by community pretty well over time.

Inside an enterprise, this enables central data team to establish best practices around SQL with proper review model themselves as well.

This is a POC (a Pre-Alpha development status as defined in https://martin-thoma.com/software-development-stages/#2-pre-alpha)

### CODE REVIEW

File: projects/vdk-plugins/vdk-smarter/src/vdk/plugin/smarter/openai_plugin_entry.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    @hookimpl
+    def vdk_initialize(self, context: CoreContext) -> None:
+        openai.api_key = context.configuration.get_value("openai_api_key")
+        self._review_enabled = context.configuration.get_value("openai_review_enabled")
+        self._openai_model = context.configuration.get_value("openai_model")
+
+    def _review_sql_query(self, sql_query: str):
+        # Refine the prompt and make configurable
+        prompt = (
+            """Using your extensive knowledge of Impala SQL, analyze the following SQL query and provide a specific feedback.
```
Comment (DeltaMichael) [inline projects/vdk-plugins/vdk-smarter/src/vdk/plugin/smarter/openai_plugin_entry.py:59]: Is this supposed to be specific to Impala?

File: projects/vdk-plugins/vdk-smarter/src/vdk/plugin/smarter/openai_plugin_entry.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            engine=self._openai_model,
+            prompt=prompt,
+            max_tokens=1000,
+            n=1,
+            stop=None,
+            temperature=0.7,
+        )
+
+        # Extract the generated review from the response
+        review = response.choices[0].text.strip()
```
Comment (DeltaMichael) [inline projects/vdk-plugins/vdk-smarter/src/vdk/plugin/smarter/openai_plugin_entry.py:79]: What happens if there is no response?

**Review by DeltaMichael — APPROVED** [review]

**Review by duyguHsnHsn — APPROVED** [review]
Looks good to me! Shouldn't be there a VEP for this?

**Comment by tozka** [comment]
> Looks good to me! Shouldn't be there a VEP for this?

Not at this stage. This is just POC (note the development status of the plugin Pre-Alpha).

**Pull request merged**